hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -189,28 +233,33 @@ public void configure() {
 				     .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
 
 		            // sending message to HIBC for ELIG
-	            .when(simple("${in.header.messageType} == {{hibc-r15-endpoint}} || ${in.header.messageType} == {{hibc-e45-endpoint}}"))
-	                .log("the HIBC endpoint(${in.header.messageType}) is reached and message will be dispatched to message queue(ELIG).")
-                    .setBody(simple(SampleMessages.e45ResponseMessage))
-	            
-	            // sending message to HIBC for ENROL
-	            .when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
-	                .log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
-                    .setBody(simple(SampleMessages.r50ResponseMessage))
+				.when(isMessageForHIBC)
+	                .log("Processing HIBC messages. Request Queue : {{hibc.request.queue}}, ReplyQ:{{hibc.reply.queue}}")
+	                .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
+                    .bean(new PopulateJMSMessageHeader()).id("PopulateJMSMessageHeaderHIBC")
+            		.log("HIBC request message ::: ${body}")
+            		.setHeader("CamelJmsDestinationName", constant(String.format("queue:///%s?targetClient=1", hibcRequestQueue)))	           		        	
+	        		.to(hibcUrl).id("ToHIBCUrl")
+                    .log("Received response message from HIBC queue ::: ${body}")
                  
 	            // others sending to JMB
 	            .otherwise()
```

Comment:
The JMB endpoint should probably have a defined set of messages that can get sent to it. This would be the same as the other endpoints e.g. RTRANS that use a 'when' clause. The 'otherwise' section could be kept just for unexpected message types and handle them with warning or error.